src/components/MTableHeader/index.js

```diff
@@ -272,7 +272,7 @@ export function MTableHeader(props) {
         headers.push(renderActionsHeader());
       }
     }
-    if (props.hasDetailPanel) {
+    if (props.hasDetailPanel && props.options.showDetailPanelIcon) {
```

Comment:
What is the default here? We don't want to brick existing code.

Reply:
Default is true
https://github.com/material-table-core/core/blob/master/src/defaults/props.options.js#L34